src/Tools/BuildValidator/LocalReferenceResolver.cs

```diff
@@ -93,40 +106,46 @@ public void CacheNames(ImmutableArray<MetadataReferenceInfo> names)
                 {
                     // A single file name can have multiple MVID, so compare by name first then
                     // open the files to check the MVID 
-                    var potentialMatches = names.Where(m => FileNameEqualityComparer.Instance.Equals(m.FileInfo, file));
-
-                    if (!potentialMatches.Any())
-                    {
-                        continue;
-                    }
-
-                    if (GetMvidForFile(file) is not { } mvid || _cache.ContainsKey(mvid))
-                    {
-                        continue;
-                    }
-
-                    var matchedReference = potentialMatches.FirstOrDefault(m => m.Mvid == mvid);
-                    if (matchedReference.FileInfo is null)
+                    foreach (var reference in references)
                     {
-                        continue;
+                        if (reference.FileInfo.Name != file.Name)
+                        {
+                            continue;
+                        }
+
+                        if (GetMvidForFile(file) is not { } mvid)
```

Comment:
In the case `GetMvidForFile` is `null` we should probably log a warning.

Reply:
Yeah if the filename already matches the reference then it seems bad if we can't read an mvid from it.